src/basic/user-util.c

```diff
@@ -555,18 +557,11 @@ int getgroups_alloc(gid_t** gids) {
                 ngroups = getgroups(0, NULL);
                 if (ngroups < 0)
                         return -errno;
-                if (ngroups == 0)
-                        return false;
-
-                free(allocated);
-
-                p = allocated = new(gid_t, ngroups);
-                if (!allocated)
-                        return -ENOMEM;
+                if (ngroups == 0) {
+                        *ret = NULL;
```

Comment:
This is fixing a double-free bug.  This should have a mention in the commit message.

Plus, mixing bug fixes with style fixes in the same commit is bad.

Reply:
uh, this would be one if we actually had a call site that calls this function with `*ret` not initialized to NULL anyway. But afaics all calls (there's only one in fact, outside of tests) actually calls this with *ret set to NULl, always.

hence, yes this is a bug, but more a bug in the sense that our coding style wasn't followed, because we have the rule that all return params must be initialized on success, and left untouched on failure.